From: Jack Steiner <steiner@sgi.com>
Subject: x86, UV: Enable 64-bit ACPI MFCG support for SGI UV2 platform x86/uv
Patch-mainline: Not yet
References: bnc#699842, fate#311784

Enable 64-bit ACPI MFCG support for SGI UV2 platform. The check
is similar to the check on UV1. UV2 has a different oem_id
string.

Signed-off-by: Jack Steiner <steiner@sgi.com>
Link: http://lkml.kernel.org/r/20110602195943.GA27079@sgi.com
Signed-off-by: Ingo Molnar <mingo@elte.hu>
Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
---
 arch/x86/pci/mmconfig-shared.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: linux-2.6.32-SLE11-SP2/arch/x86/pci/mmconfig-shared.c
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/arch/x86/pci/mmconfig-shared.c
+++ linux-2.6.32-SLE11-SP2/arch/x86/pci/mmconfig-shared.c
@@ -535,7 +535,8 @@ int pci_mmcfg_config_num;
 
 static int __init acpi_mcfg_oem_check(struct acpi_table_mcfg *mcfg)
 {
-	if (!strcmp(mcfg->header.oem_id, "SGI"))
+	if (!strcmp(mcfg->header.oem_id, "SGI") ||
+	    		!strcmp(mcfg->header.oem_id, "SGI2"))
 		acpi_mcfg_64bit_base_addr = TRUE;
 
 	return 0;
